client/question/services/FeedbackGeneratorService.js

```diff
@@ -53,6 +73,25 @@ tie.factory('FeedbackGeneratorService', [
     var consecutiveSameRuntimeErrorCounter = 0;
 
     /**
+     * Object to hold arrays indexes (values) of available correctness feedback
+     * text as listed in CORRECTNESS_FEEDBACK_TEXT, per feedback type (keys).
+     * Used in order to avoid presenting the same feedback text consecutively.
+     * A given feedback text is not used again until all other available
+     * alternative texts are used.
+     *
+     * @type {Object.<string, Array.<string>>}
+     */
+    var availableCorrectnessFeedbackIndexes = {};
+
+    /**
+     * Index of last test suite for which correctness feedback was given. Used
+     * to determine if the user has moved on to a new test suite.
+     *
+     * @type {number}
+     */
+    var previousTestSuiteId = '';
```

Comment:
Initialize to null? Weird to have a var that's supposed to be a string end up being a number.

Also something seems wrong here. We have test suite IDs, but those are strings, and they're IDs, not numerical indexes.

Reply:
I'm not sure I follow. Am I assigning numbers to this? Does var testSuiteId = testSuites[j].getId() return a number? I thought it returns a string.

Reply:
Ignore first paragraph -- I got confused by the docstring. I think the docstring is wrong here in that it refers to a 'number' type and an index rather than ID.

Reply:
Ah! that should have been 'string'. Fixed.

Reply:
Thanks! Just to make sure you caught it, note the point about index --> ID too. They mean different things: an index is a numeric index into an array, and an ID is a unique identifier.